### PR TITLE
added field name information to exception messages

### LINKED ISSUE #4484: When inserting an object with a null required field: say which one
#### Goal

Currently, when trying to insert an object that has a `null` field which is required, the exception looks like this:

```
java.lang.IllegalArgumentException: Illegal Argument: This field is not nullable.
```
Unfortunately my object has many fields, and so it's a waste of time trying to debug what the issue is.

#### Expected Results

Please change the message to something like `The field 'fooBar' is not nullable.`

#### Version of Realm and tooling
Realm version(s): 3.1.0

Realm sync feature enabled: no

Android Studio version: 2.4 Preview 5

Which Android version and device: Emulator API level 23

### PR DESCRIPTION
**Pull request by zaki50**
fixes #4484

@realm/java

### CODE REVIEW

**Comment by zaki50** [comment]
@realm/java

File: realm/realm-library/src/main/cpp/util.hpp [hunk 1]
```diff
@@ -343,7 +343,7 @@ inline bool TypeValid(JNIEnv* env, T* pTable, jlong columnIndex, int expectColTy
     int colType = pTable->get_column_type(col);
     if (colType != expectColType) {
         realm::jni_util::Log::e("Expected columnType %1, but got %2.", expectColType, pTable->get_column_type(col));
-        ThrowException(env, IllegalArgument, "ColumnType invalid.");
+        ThrowException(env, IllegalArgument, "ColumnType of '" + ((std::string) pTable->get_column_name(col)) + "' is invalid.");
```
Comment (beeender) [inline realm/realm-library/src/main/cpp/util.hpp]: shall we use `std::string(pTable->get_column_name(col))` instead to avoid c-style casting?

Comment (zaki50) [inline realm/realm-library/src/main/cpp/util.hpp]: right!

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 * Upgraded to Realm Sync 1.5.2.
 
+### Bug Fixes
```
Comment (beeender) [inline CHANGELOG.md]: It seems the changelog is from the latest. But why didn't github show a conflict warning?

Comment (zaki50) [inline CHANGELOG.md]: I've merged latest `releases`.

**Comment by zaki50** [comment]
Thanks. I'll merge this after CI build.

**Pull request merged**